_posts/2021-05-04-how-to-update-your-ref-arch.adoc

```diff
@@ -0,0 +1,490 @@
+  terragrunt state rm "$original_addr"
+}
+----
+
+=== Step 2: Update each Terragrunt live configuration to use the Service Catalog
+
+At this point, you are ready to update each live configuration! We recommend taking a bottom-up approach for migrating
+the live configurations. That is, update live configurations that don't have any downstream dependencies first, then
+work your way up the dependency graph.
+
+This has the following advantages:
+
+* Each update is self contained. Changing the live configuration of leaf services will not affect other live
+  configurations, allowing you to continue to make changes to unmigrated live configurations.
+* The leaf nodes in the Terragrunt infrastructure graph tend to be lower risk services. That is, the higher you are in
+  the graph, there are more things that depend on that infrastructure, which by definition makes that service has a
+  larger surface area. E.g., VPC has many downstream dependencies, which means that messing it up can cause lots of
+  other services to fail.
+
+However, this does mean that you will need to go back and update previously migrated services if the upstream services
+has a change. For example, Service Catalog services sometimes have output name changes, which means that you will need
+to update the references in the downstream services when you update the service.
+
+To handle this, you can identify all the downstream services that are affected by running `terragrunt validate-all` to
+identify these broken links each time a service is updated, and fix them in the same PR.
+
+Here are the steps you can take to update a single service's live configuration:
+
+[arabic]
+. Check to make sure the service has no downstream dependencies. You can use the `graph-dependencies` command to create
+  a visual representation. In the graph, the arrow points in the direction of the dependency, with the top nodes being
+  leaf nodes and the bottom being root nodes.
+    * `terragrunt graph-dependencies | dot -Tpng > graph.png`
+    * If you get an error that `dot` is not available, install https://www.graphviz.org/download/[graphviz], which
+      installs the `dot` utility.
+
+. First, you need to ensure the module is updated to the same version used in Reference Architecture version
+  https://github.com/gruntwork-io/infrastructure-modules-multi-account-acme/releases/tag/v0.0.1-20201125[20201125].
+    * If you are running an older version, follow the migration guides referenced in
+      https://github.com/gruntwork-io/infrastructure-modules-multi-account-acme/releases[the Reference Architecture
+      releases] to update to the latest version. This is necessary because the Service Catalog module references use newer
+      versions from the Module Catalog than what is shipped with v1.0 of the Reference Architecture. Once you've upgraded to
+      https://github.com/gruntwork-io/infrastructure-modules-multi-account-acme/releases/tag/v0.0.1-20201125[20201125],
```

Comment:
this is not the latest version. there are 2 releases after this. what if we are on a later version?

Reply:
Newer is fine. Updating the text to reflect that. Done in a7b778a6..549c027e.